### ISSUE TITLE
Feedback: It should be possible to request readwrite access before getting read access.

### ISSUE THREAD

**Issue by tormodAase** [WICG/file-system-access#302]
There's no way to send only one permission request if you need write access. That is annoying. There should be a way to define which permissions you need in the showDirectoryPicker/showFilePicker methods, reducing permission requests by 50% (from 2 to 1) when you need write access.

### Example:
This code sends two requests to the user. First a read permission request, followed by a write permission request. 
```
window.showDirectoryPicker().then(handle => {
  handle.requestPermission({mode:'readwrite'})
})
```

If there were some way to define which permissions you need in the showDirectoryPicker method, we could reduce the # of requests to 1:
```
window.showDirectoryPicker({mode:'readwrite'}).then(handle => {
  //some code
})
```

**Comment by a-sully** [comment]
Duplicate of #89 

This is in the works, though! See #300

**Comment by tomayac** [comment]
While a `mode` option was added to `showDirectoryPicker()`, it's not possible to do the same with `showOpenFilePicker()`, which the original post [mentioned](https://github.com/WICG/file-system-access/issues/302#:~:text=in%20the%20showDirectoryPicker/-,showFilePicker,-methods%2C%20reducing%20permission), but the Issue was then merged with #89, which only looked at the `showDirectoryPicker()` part.

This came up now again [in the context of the Intent to Ship for Android](https://groups.google.com/a/chromium.org/g/blink-dev/c/x3IcFv2jY6c/m/HxuEfzBHAAAJ#:~:text=There%20is%20one,as%20many%20providers%29.). 

CC: @joelhockey
